mozci/push.py

```diff
@@ -1030,3 +1030,72 @@ def make_push_objects(**kwargs):
             cur._child = pushes[i + 1]
 
     return pushes
+
+
+def make_summary_objects(**kwargs):
+    func = "make_{}_summary_objects".format(kwargs.get("type").lower())
+
+    # Obtain list of all pushes for the specified branch, from_date and to_date.
+    pushes = make_push_objects(**kwargs)
+
+    try:
+        # Dynamically run the method name `func`.
+        summaries = globals()[func](pushes, **kwargs)
+    except KeyError:
+        return []
+
+    # Sort by either the label or the name attribute.
+    # sort_attr = 'label' if kwargs.get('type').lower() == 'label' else 'name')
+    summaries.sort(key=lambda x: (getattr(x, "label", "name")))
+    return summaries
+
+
+def make_label_summary_objects(pushes, **kwargs):
```

Comment:
Would it be possible to turn these two function into private methods, and then have `make_summary_objects` as the single public entry point for this? I think it might make for a cleaner API.

Reply:
To clarify, making these into a private method attached to the respective classes?